sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnVector.java

```diff
@@ -367,9 +551,13 @@ public Object get(int ordinal, DataType dataType) {
   /**
    * Returns the array at rowid.
    */
-  public final ColumnVector.Array getArray(int rowId) {
-    resultArray.length = getArrayLength(rowId);
-    resultArray.offset = getArrayOffset(rowId);
+  public final ArrayData getArray(int rowId) {
```

Comment:
We should not change the return type. `ColumnVector` will be public eventually, and `ArrayData` is not a public type.

Reply:
I see.
One question. `ColumnVector.Array` has some public fields such as `length`. I think that it would be good to use an accessor `length` or `getLength`. What do you think?